### ISSUE TITLE
Mastodon `replies` can be fetched correctly

### ISSUE THREAD

**Issue by goffi-contrib** [mastodon/mastodon#17442]
### Steps to reproduce the problem

Hello,

for the context I'm currently working on [ActivityPub <=> XMPP gateway](https://nlnet.nl/project/Libervia/), and I need to access replies to an item to construct corresponding XMPP object.

Unfortunately it seems that `replies` object as retrieved on Mastodon are not compliant with [ActivityStream Collection Paging](https://www.w3.org/TR/activitystreams-core/#collections) and thus it is not possible to retrieve all items.

For instance, on item at https://mastodon.social/@Goffi/106114174053289219, the `replies` object looks like:

```json
"replies": {
    "first": {
        "items": [],
        "next": "https://mastodon.social/users/Goffi/statuses/106114174053289219/replies?only_other_accounts=true&page=true",
        "partOf": "https://mastodon.social/users/Goffi/statuses/106114174053289219/replies",
        "type": "CollectionPage"
    },
    "id": "https://mastodon.social/users/Goffi/statuses/106114174053289219/replies",
    "type": "Collection"
}
```

There are a few issues with that:

1. I would expect an `OrderedCollection` here, not `Collection`
2. `totalItems` is missing
3. `last` page is missing
4. the collection here seems empty, but if we get next page with `next`, we get some replies. However, using next again will link back to the same page, resulting in an infinite loop. The behaviour is the same with other objects where `items` is not empty.
5. there is never a `prev` field

As a result, there is no way to know when the collection is finished, or to get pages in reverse order.

I realise that Mastodon is only getting new items after somebody is following an actor, but as `replies` are available, is would be good to be able to get all items from this collection.

### Expected behaviour

`replies` field should follow https://www.w3.org/TR/activitystreams-core/#collections, `orderedCollection` should be used, `totalItems` should be present, `last` page should be present, `prev` should be present when relevant, `next` should not link back an previous page and thus avoid a loop

### Actual behaviour

Wrong collection type, `totalItems` missing, `last` page missing, `next` leading to infinite loop, `prev` missing

### Specifications

This can be seen on mastodon.social, I'm seeing also this locally on Mastodon c5606db9dae58aec155751105e58d85170efef98

Thanks!

**Comment by ClearlyClaire** [comment]
1. I'm not sure why it would be an `OrderedCollection` over a `Collection`; the ordering here is tricky as it first returns self-replies, the primary reason for that feature being fetching “threads” (as in, a chain of self-replies), not the full replies list (although it's made so that it should be possible)
2. will have to get a look, but I think it was intentional to leave that out, out of performance concerns, but I may misremember
3. having support for `last` and backwards traversal of the replies would needlessly complicate the code, and may also have performance issues
4. see 1. for the empty first page and the use of `next`; `next` eventually looping is definitely a bug though and I'll have a look next week
5. see 3.

**Comment by nightpool** [comment]
The empty first page is kind of confusing, can't we just do the same thing
we do with followers and make first an href instead of next?

**Comment by ClearlyClaire** [comment]
The looping `next` should be addressed by #17462

> The empty first page is kind of confusing, can't we just do the same thing we do with followers and make first an href instead of next?

I'm not sure what you mean by that. The first page is not always empty, it's just that the first page is dedicated to self-replies, so it can be empty without the whole collection being empty. We could point to the second page as `first` if the first one is empty, but then it would make Mastodon perform useless extra queries when discovering a toot with a replies collection (currently, Mastodon checks the first page and only the first page, for the sole purpose of fetching self-replies in threads).

**Comment by goffi-contrib** [comment]
Thanks for the quick answer and fix.

 1 . it just feels more coherent with collections like `inbox` or `outbox` which are `OrderedCollection` (and would make it easier to use), but AFAIK nothing impose it, so it's understandable to use unordered `Collection` in case of technical constraints
2,3,4,5. `totalItem`, `last` and backward pagination would have been helpful to me, but [ActivityStream](https://www.w3.org/TR/activitystreams-core/#collections) doesn't mandate them, thus I'll deal without it.

This issue is now fine by me thanks to your fix, I'm closing.
